lib/saluki-components/src/sources/checks/mod.rs

```diff
@@ -315,7 +319,23 @@ async fn process_listener(source_context: SourceContext, listener_context: liste
                 info!("Running a check request: {check_request}");
 
                 match scheduler.run_check(check_request) {
-                    Ok(_) => {}
+                    Ok(_) => {
+                        // drain the global queue and send everything to source_context
+                        let mut event_buffer = source_context.event_buffer_pool().acquire().await;
+
+                        let mut q = aggregator::SUBMISSION_QUEUE.lock().unwrap();
+                        debug!("will send {} metrics to the event buffer", q.size());
+                        while q.size() > 0 {
+                            let check_metric = q.remove().expect("can't read from the global submission queue");
+
+                            // TODO(remy): fit the the deserializer/codec architecture instead of doing it here
+                            let event: Event = aggregator::check_metric_as_event(check_metric).expect("can't convert");
+                            event_buffer.push(event);
+                            debug!("one metric sent to the event buffer from check execution");
+                        }
+                        debug!("queue drained");
+                        drop(q); // we're in an infinite loop, explicitely drop q to unlock the mutex
```

Comment:
Another benefit of switching to `VecDeque` instead of that queue type would be that you could reduce this code down to:

```suggestion
                        let mut q = aggregator::SUBMISSION_QUEUE.lock().unwrap();
                        event_buffer.extend(q.drain(..).filter_map(aggregator::check_metric_as_event));
                        drop(q);
```

You'd have to change `check_metric_to_event` to return `Option<Event>`, though.... but that might not be a bad change overall.